### PR TITLE
Use a shorter data window for recording and inference

### PR DESCRIPTION
**Pull request by microbit-robert**
We now record data for one second (990ms) and infer using time window while maintaining backwards compatibility with existing data samples that use the legacy window of 1800 ms. This makes the model feel much more responsive to user gesture/action changes.

- Look at the user's existing gestures to determine whether we should use the legacy data window
- Default to the new shorter data window where possible (new session, user deletes all recordings, user deletes all actions)
- Maintain existing data window for dataset.json and hex file loads

Other fixes and upgrades:
- Upgrade pxt-microbit-ml extension to include ZCR fix
- Fix prop casing for AlertIcon
- Remove circular reference in tour files
- Pull up listener for MakeCode project load so it affects all pages

### CODE REVIEW

File: src/makecode/generate-custom-scripts.ts [hunk 1]
```diff
@@ -48,8 +48,8 @@ const arrayBufferToHexString = (input: Uint8Array): string =>
 
 export const getAutogeneratedTs = (gs: GestureData[], m: LayersModel) => {
   const customHeaderBlob = generateBlob({
-    samples_period: 25,
-    samples_length: 80,
+    samples_period: 20, // Default value for accelerometer period.
+    samples_length: 50, // Number of samples required at 20 ms intervals for 1 second of data.
```
Comment (microbit-matt-hillsdon) [inline src/makecode/generate-custom-scripts.ts:52]: We think this needs to be parameterised for legacy projects.

File: src/store.ts [hunk 2]
```diff
@@ -758,6 +792,21 @@ const createMlStore = (logging: Logging) => {
 
 export const useStore = createMlStore(deployment.logging);
 
+const getDataWindowFromGestures = (gestures: GestureData[]): DataWindow => {
+  const dataLength = gestures.flatMap((g) => g.recordings)[0]?.data.x.length;
+  return dataLength > legacyDataWindow.minSamples
```
Comment (microbit-matt-hillsdon) [inline src/store.ts:797]: Should this be >= ?

**Comment by microbit-matt-hillsdon** [comment]
Otherwise this is looking good.

File: src/store.ts (hunk 2 above)
Comment (microbit-robert) [reply src/store.ts]: Yes [c35545e](https://github.com/microbit-foundation/ml-trainer/pull/431/commits/c35545ee4b3d7b771344601b6542fe857c018fb0)

File: src/makecode/generate-custom-scripts.ts (hunk 1 above)
Comment (microbit-robert) [reply src/makecode/generate-custom-scripts.ts]: Done here, a bit more change than  I hoped [dededc1](https://github.com/microbit-foundation/ml-trainer/pull/431/commits/dededc1a2265f9fff4215c5539287a1cba8e558d)

**Review by microbit-matt-hillsdon — APPROVED** [review]

**Pull request merged**